bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/StringListType.java

```diff
@@ -37,27 +36,25 @@ public StringListType() {
         typeDetails = Collections.emptyList();
     }
 
+    public StringListType(List<String> rows) {
+        this.typeDetails = new ArrayList<>(rows);
+    }
+
     public StringListType(StringType... rows) {
-        typeDetails = new ArrayList<String>(rows.length);
-        for (StringType row : rows) {
-            typeDetails.add(row.toString());
-        }
+        this(Arrays.stream(rows)
+            .map(StringType::toString)
+            .collect(Collectors.toList()));
     }
 
     public StringListType(String... rows) {
-        typeDetails = Arrays.asList(rows);
+        this(Arrays.asList(rows));
```

Comment:
What I don't like about this approach is that we are now creating a collection twice. Therefore I'd prefer the previous approach with directly assigning the newly created collection to `typeDetails`.